File: spring/core/DependencyInjectionViaGroovy/src/main/java/com/instil/di/groovy/beans/PaymentEngineMock.java
package com.instil.di.groovy.beans;

import com.instil.di.groovy.contracts.PaymentEngine;

import java.util.List;

public class PaymentEngineMock implements PaymentEngine {
    private final List<String> bannedCards;

    public PaymentEngineMock(List<String> bannedCards) {
        super();
        this.bannedCards = bannedCards;
    }

    public boolean authorize(String cardNo, double amount) {
        if (bannedCards.contains(cardNo)) {
            return false;
        }
        return amount < 1000;
    }
}

File: spring/core/KotlinBasedDI/src/main/kotlin/com/instil/PaymentEngineMock.kt
package com.instil

class PaymentEngineMock(private val bannedCards: List<String>) : PaymentEngine {
	override fun authorize(cardNo: String, amount: Double): Boolean {
		println("\tAuthorizing payment...")
		return if(bannedCards.contains(cardNo)) false else amount < 1000
    }
}
